spec/unit/recipes/plugins_spec.rb

```diff
@@ -0,0 +1,58 @@
+#
+# Cookbook Name:: dokku
+# Spec:: plugins
+#
+# Copyright (c) 2015 Nick Charlton, MIT licensed.
+
+require "spec_helper"
+
+describe "dokku::plugins" do
+  context "when all attributes are default" do
+    let(:chef_run) do
+      runner = ChefSpec::ServerRunner.new
+      runner.converge(described_recipe)
+    end
+
+    it "does nothing" do
+      expect { chef_run }.to_not raise_error
+    end
+  end
+
+  context "when plugins exist in attributes and no actions are listed" do
+    let(:chef_run) do
+      runner = ChefSpec::ServerRunner.new do |node|
+        node.set["dokku"]["plugins"] = [{
+          name: "redis",
+          url: "https://github.com/dokku/dokku-redis.git"
+        }]
+      end
+      runner.converge(described_recipe)
+    end
+
+    it "installs plugins" do
+      expect(chef_run).to install_dokku_plugin(
+        "redis").with(url: "https://github.com/dokku/dokku-redis.git")
+    end
+  end
+
+  context "when plugins exist in attributes" do
+    let(:chef_run) do
+      runner = ChefSpec::ServerRunner.new do |node|
+        node.set["dokku"]["plugins"] = [
+          { name: "redis", url: "https://github.com/dokku/dokku-redis.git",
+            action: "install" },
+          { name: "mongo", url: "https://github.com/dokku/dokku-mongo.git",
+            action: "uninstall" },
+          ]
```

Comment:
Indent the right bracket the same as the start of the line where the left bracket is.